scripts/python-skywater-pdk/skywater_pdk/fet_simulator/fet_simulator.py

```diff
@@ -0,0 +1,293 @@
+        pd="'2*(W+0.29)'", as_="'W*0.29'", ps="'2*(W+0.29)'", nrd="'0.29/W'",
+        nrs="'0.29/W'", sa=0, sb=0, sd=0, nf=1, mult=1
+    )
+    return c
+
+
+def run_sim(c, iparam, fet_W):
+    """
+    Runs simulation for a given circuit with FET cell.
+
+    Parameters
+    ----------
+    c: circuit to simulate
+    iparam:
+        template string for creating internal parameter names (gm, id, gds,
+        cgg) for a given FET
+    fet_W: FET width
+    """
+    sim = c.simulator()
+    sim.save_internal_parameters(
+        iparam % 'gm', iparam % 'id', iparam % 'gds', iparam % 'cgg'
+    )
+
+    try:
+        # run the dc simulation
+        an = sim.dc(Vgg=slice(0, 1.8, 0.01))
+
+        # calculate needed values..need as_ndarray() since most of these have
+        # None as the unit and that causes an error
+        gm_id = (an.internal_parameters[iparam % 'gm'].as_ndarray() /
+                 an.internal_parameters[iparam % 'id'].as_ndarray())
+        ft = (an.internal_parameters[iparam % 'gm'].as_ndarray() /
+              an.internal_parameters[iparam % 'cgg'].as_ndarray())
+        id_W = (an.internal_parameters[iparam % 'id'].as_ndarray() / fet_W)
+        gm_gds = (an.internal_parameters[iparam % 'gm'].as_ndarray() /
+                  an.internal_parameters[iparam % 'gds'].as_ndarray())
+    except Exception:
+        sim.ngspice.destroy('all')
+        sim.ngspice.reset()
+        raise
+
+    sim.ngspice.destroy('all')
+    sim.ngspice.reset()
+
+    return gm_id, ft, id_W, gm_gds, an.nodes['v-sweep']
+
+
+def init_plots(fet_name, W):
+    """
+    Initializes plots for FET bins simulation.
+
+    Parameters
+    ----------
+    fet_name: name of the current FET cell
+    W: FET width
+    """
+    figs = [plt.figure(), plt.figure(), plt.figure(), plt.figure()]
+    plts = [f.subplots() for f in figs]
+    figs[0].suptitle(f'{fet_name} Id/W vs gm/Id (W = {W})')
+    plts[0].set_xlabel("gm/Id")
+    plts[0].set_ylabel("Id/W")
+    plts[0].grid(True)
+    figs[1].suptitle(f'{fet_name} fT vs gm/Id (W = {W})')
+    plts[1].set_xlabel("gm/Id")
+    plts[1].set_ylabel("f_T")
+    plts[1].grid(True)
+    figs[2].suptitle(f'{fet_name} gm/gds vs gm/Id (W = {W})')
+    plts[2].set_xlabel("gm/Id")
+    plts[2].set_ylabel("gm/gds")
+    plts[2].grid(True)
+    figs[3].suptitle(f'{fet_name} gm/Id vs Vgg (W = {W})')
+    plts[3].set_xlabel("Vgg")
+    plts[3].set_ylabel("gm/Id")
+    plts[3].grid(True)
+    return figs, plts
+
+
+def gen_plots(gm_id, id_W, ft, gm_gds, vsweep, fet_W, fet_L, plts):
+    """
+    Generates plot lines for FET bins simulation parameters.
+
+    Parameters
+    ----------
+    gm_id: gm/Id values
+    id_W: Id/W values
+    ft: f_T values
+    gm_gds: gm/gds values
+    vsweep: v-sweep values
+    fet_W: FET width
+    fet_L: FET length
+    plts: plots on which plot lines should be drawn
+    """
+    # plot some interesting things
+    plts[0].plot(gm_id, id_W, label=f'W {fet_W} x L {fet_L}')
```

Comment:
Maybe `plots` is a better name than `plts`. You are only saving one letter and it took me a second to figure out what this meant.

Reply:
It is also very close to the global module `plt`.

Reply:
fixed